packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,739 @@
+			}
+			break;
+		case TreeShape.Wide:
+			assert(isUnwrappedNode(trees[0].root));
+			for (let i = 0; i < numberOfNodes - 1; i++) {
+				trees[0].root[getField](localFieldKey).insertNodes(
+					i,
+					singleTextCursor(generateTreeData(dataType, random)),
+				);
+			}
+			break;
+		default:
+			unreachableCase(shape);
+	}
+	return trees[0];
+}
+const booleans = [true, false];
+
+function generateTreeData(dataType: TestPrimitives, random: IRandom): JsonableTree {
+	const insertValue = random.integer(Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER);
+	let map;
+	switch (dataType) {
+		case TestPrimitives.Number:
+			return { value: insertValue, type: dataSchema.name };
+		case TestPrimitives.Float:
+			return { value: insertValue, type: dataSchema.name };
+		case TestPrimitives.String:
+			return {
+				value: random.real(0, Number.MAX_SAFE_INTEGER).toString(),
+				type: dataSchema.name,
+			};
+		case TestPrimitives.Boolean:
+			return { value: random.pick(booleans), type: dataSchema.name };
+		case TestPrimitives.Map:
+			map = {
+				mapField2: {
+					mapField3: [{ type: dataSchema.name, value: insertValue.toString() }],
+				},
+			};
+			return {
+				value: map,
+				type: dataSchema.name,
+			};
+		default:
+			unreachableCase(dataType);
+	}
+}
+
+function getTestTreeAsJSObject(
+	numberOfNodes: number,
+	shape: TreeShape,
+	dataType: TestPrimitives,
+): any {
+	const seed = 0;
+	const random = makeRandom(seed);
+	let tree;
+	switch (shape) {
+		case TreeShape.Deep:
+			tree = [getJSTestTreeDeep(numberOfNodes, dataType, random)];
+			break;
+		case TreeShape.Wide:
+			tree = getJSTestTreeWide(numberOfNodes, dataType, random);
+			break;
+		default:
+			unreachableCase(shape);
+	}
+	const testTreeJS = JSON.parse(JSON.stringify(tree));
+	return testTreeJS;
+}
+
+function getJSTestTreeWide(numberOfNodes: number, dataType: TestPrimitives, random: IRandom): any {
+	const nodes = [];
+	for (let i = 0; i < numberOfNodes - 1; i++) {
+		const node = generateTreeData(dataType, random);
+		nodes.push(node);
+	}
+	const tree = {
+		type: dataSchema.name,
+		fields: {
+			foo: nodes,
+		},
+		value: generateTreeData(dataType, random).value,
+	};
+	return tree;
+}
+
+function getJSTestTreeDeep(numberOfNodes: number, dataType: TestPrimitives, random: IRandom): any {
+	const node = generateTreeData(dataType, random);
+	if (numberOfNodes === 1) {
+		return {
+			type: dataSchema.name,
+			value: node.value,
+		};
+	}
+	const tree = {
+		type: dataSchema.name,
+		fields: {
+			foo: [getJSTestTreeDeep(numberOfNodes - 1, dataType, random)],
+		},
+		value: node.value,
+	};
+	return tree;
+}
+
+function readTreeAsJSObject(tree: any) {
+	for (const key of Object.keys(tree)) {
+		if (typeof tree[key] === "object" && tree[key] !== null) readTreeAsJSObject(tree[key]);
+		if (key === "value") {
+			assert(tree[key] !== undefined);
+		}
+	}
+}
+
+function manipulateTreeAsJSObject(tree: any, dataType: TestPrimitives, random: IRandom): any {
```

Comment:
this does not appear to have a return value, so its return type should reflect that.
Also don't use `any`.
Also needs more documentation.

Is this used for the JS object references tests? Is it also used for the EditableTree tests? It would be awesome to run the same code over both if possible for comparison. Right now its hard to tell which of these its doing.

Reply:
added documentation, removed any type. This function is currently used to change the leaf node of the tree when represented as a JS Object.